privacyidea/lib/eventhandler/base.py

```diff
@@ -311,12 +335,30 @@ def check_condition(self, options):
             #  the user has only one token
             token_obj_list = get_tokens(user=user)
         if len(token_obj_list) == 1:
+            # There is a token involved, so we determine it's resolvers and realms
             token_obj = token_obj_list[0]
             tokenrealms = token_obj.get_realms()
             tokentype = token_obj.get_tokentype()
 
-        if "realm" in conditions:
-            if user.realm != conditions.get("realm"):
+            all_realms = get_realms()
+            for tokenrealm in tokenrealms:
+                resolvers = all_realms.get(tokenrealm, {}).get("resolver")
```

Comment:
Is there any situation in which there is no ``"resolver"`` key in the dictionary (e.g. if ``all_realms`` does not contain ``tokenrealm``?)? In this case, the line below would fail with ``AttributeError: 'NoneType' object has no attribute 'get'``

Should we maybe change this to ``.get("resolver", {})``?

Reply:
I am not sure, if

    .get(tokenrealm, {}) 

returns {}.
But I see where you are aiming at. It will fail in ``for  r in resolvers`` in case of None.

Will do!

Reply:
Done in 3fdefe23

Reply:
Thanks!